### PR TITLE
Update babel parser

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Error Codes are introduced in babel/parser since Babel 7.14.0. Also almost error messages throw by babel/parser are fixed for consistent.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/format/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -367,6 +367,17 @@ console.log(Bork.staticFunction()); // > "babelIsCool"
 ================================================================================
 `;
 
+exports[`class-static-block.js [babel-ts] format 1`] = `
+"Unexpected token (4:10)
+  2 |   static #x = 42;
+  3 |   static y;
+> 4 |   static {
```
Comment (sosukesuzuki) [inline tests/format/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:374]: This should be fixed by babel/parser.

Comment (sosukesuzuki) [reply tests/format/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:374]: https://github.com/babel/babel/issues/13234

File: src/language-js/parser-babel.js [hunk 2]
```diff
@@ -145,71 +142,61 @@ const parseTypeScript = createParse(
...
-    /^Private name #.*? is not defined$/.test(message) ||
-    /^`.*?` has already been exported\. Exported identifiers must be unique\.$/.test(
-      message
-    )
-  ) {
-    return false;
-  }
 
-  return true;
+function shouldRethrowRecoveredError(error) {
```
Comment (fisker) [inline src/language-js/parser-babel.js:198]: Remove this function, use `allowedMessageCodes` directly.

File: src/language-js/parser-babel.js [hunk 3]
```diff
@@ -24,13 +24,10 @@ const parseOptions = {
     // When adding a plugin, please add a test in `tests/js/babel-plugins`,
     // To remove plugins, remove it here and run `yarn test tests/js/babel-plugins` to verify
     "doExpressions",
-    "classProperties",
     "exportDefaultFrom",
     "functionBind",
     "functionSent",
-    "classPrivateProperties",
     "throwExpressions",
-    "classPrivateMethods",
```
Comment (fisker) [inline src/language-js/parser-babel.js:33]: Can `moduleStringNames` plugin removed?

File: src/language-js/parser-babel.js [hunk 4]
```diff
@@ -145,71 +142,61 @@ const parseTypeScript = createParse(
...
-  "A type-only import can specify a default import or named bindings, but not both.",
-  "An implementation cannot be declared in ambient contexts.",
-  "Classes may not have a field named 'constructor'",
+// Error codes are defined in
+//  - https://github.com/babel/babel/blob/v7.14.0/packages/babel-parser/src/parser/error-message.js
+//  - https://github.com/babel/babel/blob/v7.14.0/packages/babel-parser/src/plugins/typescript/index.js#L69-L153
+//  - https://github.com/babel/babel/blob/v7.14.0/packages/babel-parser/src/plugins/flow/index.js#L51-L140
+//  - https://github.com/babel/babel/blob/v7.14.0/packages/babel-parser/src/plugins/jsx/index.js#L23-L39
+const allowedMessageCodes = new Set([
+  "StrictNumericEscape",
```
Comment (fisker) [inline src/language-js/parser-babel.js:151]: The error codes seem less readable than the message. 😢

File: src/language-js/parser-babel.js (hunk 3 above)
Comment (fisker) [reply src/language-js/parser-babel.js:33]: https://github.com/babel/babel/pull/13195

File: src/language-js/parser-babel.js (hunk 4 above)
Comment (sosukesuzuki) [reply src/language-js/parser-babel.js:152]: Probably these error codes are used by only us now, so we can fix too less readable messages on Babel side.

File: tests/format/misc/errors/js/reserved-word/jsfmt.spec.js [hunk 5]
```diff
@@ -5,24 +5,3 @@ run_spec(
...
-      "interface();",
-      "class interface {}",
-      "interface ? true : false;",
-      "function interface() {}",
-      'import interface from "foo";',
-      "interface.foo;",
-      "new interface();",
-      '(interface, "foo");',
-      "void interface;",
-      'const interface = "foo";',
```
Comment (sosukesuzuki) [inline tests/format/misc/errors/js/reserved-word/jsfmt.spec.js:24]: This change makes the Reserved word check rough...

**Review by fisker — APPROVED** [review]

**Pull request merged**